### PR TITLE
[MINI-2947] improvement: migrate to file verification for manifest

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Manifest verification has been added previously in https://github.com/rakutentech/android-miniapp/pull/305 where `MiniAppManifestVerifier` was introduced. After finishing of MINI-2945, this PR aims to refactor the verification process with manifest files. Mainly, `MiniAppManifestVerifier` will use the functions of `StoreHashVerifier` same like `CachedMiniAppVerifier`.

## Links
- MINI-2947

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/storage/verifier/StoreHashVerifierSpec.kt [hunk 1]
```diff
@@ -3,41 +3,75 @@ package com.rakuten.tech.mobile.miniapp.storage.verifier
...
+import org.amshove.kluent.*
+import org.junit.Rule
 import org.junit.Test
+import org.junit.rules.TemporaryFolder
 import org.junit.runner.RunWith
+import java.io.File
 
 @RunWith(AndroidJUnit4::class)
 @ExperimentalCoroutinesApi
 class StoreHashVerifierSpec {
```
Comment (khairul-alam-licon) [inline miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/storage/verifier/StoreHashVerifierSpec.kt:18]: Unit test cases has been restored in this PR what was added previously in https://github.com/rakutentech/android-miniapp/pull/153

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/verifier/CachedMiniAppVerifier.kt [hunk 2]
```diff
@@ -1,50 +1,17 @@
...
     fun verify(appId: String, directory: File): Boolean =
-        storeHashVerifier.verify(appId, calculateHash(directory))
+        storeHashVerifier.verify(appId, directory)
 
     /** Stores hash in encrypted shared preferences. */
     suspend fun storeHashAsync(appId: String, directory: File) =
-        storeHashVerifier.storeHashAsync(appId, calculateHash(directory))
-
-    @SuppressWarnings("TooGenericExceptionCaught", "LongMethod", "NestedBlockDepth")
-    private fun calculateHash(
```
Comment (khairul-alam-licon) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/verifier/CachedMiniAppVerifier.kt:22]: It's moved to `StoreHashVerifier`.

**Review by corycaywood — APPROVED** [review]
LGTM 👍

**Review by munir-rakuten — APPROVED** [review]
looks good

**Pull request merged**